api/v1alpha4/nutanixmachine_types.go

```diff
@@ -61,13 +61,12 @@ type NutanixMachineSpec struct {
 	// of the Prism Central), in which the Machine's VM will be created.
 	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
 	// or using the prism_central API.
-	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Optional
 	Cluster NutanixResourceIdentifier `json:"cluster"`
 	// subnet is to identify the cluster's network subnet to use for the Machine's VM
 	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
 	// or using the prism_central API.
-	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:Optional
```

Comment:
lets add comment why is this optional and if missing what else is required

Reply:
We need it for backward compatibility. This attribute should only be defined if failure domains are not configured on the NutanixCluster object.